bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/gemini/gems/Canopus.java

```diff
@@ -227,154 +145,99 @@ public GuideStarValidation validate(SPTarget guideStar, ObsContext ctx) {
-            ).getOrElse(GuideStarValidation.UNDEFINED);
-        }
-
-            // Returns true if the area of the probe arm for the given wfs contains the
-        // coordinates of the given guide star
-        // (i.e.: The guide star is vignetted by the wfs probe arm)
-        private static Option<Boolean> isVignetted(Wfs wfs, SPTarget guideStar, ObsContext ctx) {
-            return guideStar.getSkycalcCoordinates(ctx.getSchedulingBlockStart()).flatMap(gscoords ->
-                    ctx.getBaseCoordinates().flatMap(coords ->
-                    wfs.probeArm(ctx, false).map(a -> {
-                        if (a == null) return false;
-                        CoordinateDiff diff = new CoordinateDiff(coords, gscoords);
-                        Offset dis = diff.getOffset();
-                        double p = -dis.p().toArcsecs().getMagnitude();
-                        double q = -dis.q().toArcsecs().getMagnitude();
-                        return a.contains(p, q);
-                    })));
-        }
-
         /**
          * Check if the primary guide star is in range from the given offset
          *
          * @param ctx ObsContext to get guide star and base coordinates from.
          * @param offset to check if the guide star is in range
          * @return true if guide star is in range from the given offset, false otherwise
          */
-        public boolean inRange(ObsContext ctx, Offset offset){
-            Option<GuideProbeTargets> gptOpt = ctx.getTargets().getPrimaryGuideProbeTargets(this);
-            if (gptOpt.isEmpty()) return true;
-            GuideProbeTargets gpt = gptOpt.getValue();
-
-            Option<SPTarget> guideStarOpt = gpt.getPrimary();
-            if (guideStarOpt.isEmpty()) return true;
-            SPTarget guideStar = guideStarOpt.getValue();
-
-            // Calculate the difference between the coordinate and the observation's base position.
-            return guideStar
-                .getSkycalcCoordinates(ctx.getSchedulingBlockStart())
-                .flatMap(gscoords -> ctx.getBaseCoordinates()
-                    .map(base -> {
-                        CoordinateDiff diff = new CoordinateDiff(base, gscoords);
-                        // Get offset and switch it to be defined in the same coordinate
-                        // system as the shape.
-                        Offset dis = diff.getOffset();
-                        double p = -dis.p().toArcsecs().getMagnitude();
-                        double q = -dis.q().toArcsecs().getMagnitude();
-                        Set<Offset> offsets = new TreeSet<>();
-                        offsets.add(offset);
-
-                        Area a = Canopus.instance.offsetIntersection(ctx, offsets);
-                        return a != null && a.contains(p, q);
-                    }))
-                .getOrElse(false);
+        public boolean inRange(final ObsContext ctx, final Offset offset) {
```

Comment:
This just cleans up the code to use `Option`s more cleanly instead of having to do calls like `opt.isEmpty` and `opt.getValue`.